tests/test_cliCmdQueryDatasetTypes.py

```diff
@@ -0,0 +1,66 @@
+# This file is part of daf_butler.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Unit tests for daf_butler CLI query-collections command.
+"""
+
+import click
+import unittest
+import yaml
+
+from lsst.daf.butler import Butler, DatasetType, StorageClass
+from lsst.daf.butler.cli.butler import cli
+from lsst.daf.butler.tests.mockeredTest import MockeredTestBase
+
+
+class QueryDatasetTypesCmdTest(MockeredTestBase):
+
+    defaultExpected = dict(repo=None)
+
+    def test_all(self):
+        """Test the required parameter.
+        """
+        self.run_test(["query-dataset-types", "here"],
+                      self.makeExpected(repo="here"))
+
+    def test_requiredMissing(self):
+        """Test that if the required parameter is missing it fails"""
+        self.run_missing(["query-dataset-types"], 'Error: Missing argument "REPO".')
+
+
+class QueryDatasetTypesScriptTest(unittest.TestCase):
+
+    def testQueryDatasetTypes(self):
+        expected = {"dataset-types": ["test"]}
+        runner = click.testing.CliRunner()
+        with runner.isolated_filesystem():
+            Butler.makeRepo("here")
+            butler = Butler("here", writeable=True)
```

Comment:
Again I have a slight preference for the return value of makeRepo being used here.